content/src/controller.js

```diff
@@ -2009,6 +2018,55 @@ function sortByName(a, b) {
 function renderDirectory(position) {
   var pane = paneatpos(position);
   var mpp = model.pane[pane];
+  mpp.bydate = defaultDirSortingByDate();
+  var filename = mpp.filename;
+  view.setPaneLinkText(pane, getUpdatedLinksArray(pane), filename, model.ownername);
+  updateTopControls(false);
+}
+
+
+function updateSortResults(pane) {
+  view.setPaneLinks(pane,getUpdatedLinksArray(pane));
+  updateTopControls(false);
+}
+
+function updateSearchResults(pane, search, cb) {
+  var mpp = model.pane[pane];
+  if(specialowner()) {
+    var cacheResults = cache.get('search-keys', search);
+    if(cacheResults) {
+      mpp.data = cacheResults;
+      view.setPaneLinks(pane, getUpdatedLinksArray(pane));
+      cb && cb();
+    } else {
+      storage.loadFile(model.ownername, mpp.filename+"?prefix="+search, true, function(m) {
```

Comment:
We can refresh the cache when navigating to a directory listing. What do you think